### PR TITLE
Feature/detail

### PR DESCRIPTION
**Pull request by ArtemisSoftware**

### CODE REVIEW

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/DetailState.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import com.artemissoftware.tasky.agenda.AgendaItemType
+import com.artemissoftware.tasky.agenda.domain.models.AgendaItem
+import java.time.LocalDate
+import java.time.LocalTime
+
+data class DetailState(
+    val isLoading: Boolean = false,
+    val isEditing: Boolean = false,
+    val agendaItemType: AgendaItemType? = null,
```
Comment (MKiperszmid) [inline app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/DetailState.kt:11]: Wouldn't this be covered in the "AgendaItem"?

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/DetailState.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    val agendaItem: AgendaItem? = null,
+    val startDate: LocalDate = LocalDate.now(),
+    val endDate: LocalDate = LocalDate.now(),
+    val endTime: LocalTime = LocalTime.now(),
+    val startTime: LocalTime = LocalTime.now(),
```
Comment (MKiperszmid) [inline app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/DetailState.kt:16]: You can use LocalDateTime to handle start date and time in one variable, and end date and time in another variable

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/DetailScreen.kt [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+                                    modifier = Modifier.fillMaxWidth(),
+                                    notificationOptions = emptyList(), //TODO: replace with data form the database when viewmodel is ready
+                                    onNotificationSelected = {
+                                        events(DetailEvents.UpdateNotification(it))
+                                    }
+                                )
+
+                                TaskyDivider(top = 20.dp, bottom = 30.dp)
+
+                                if(type is AgendaItemType.Event) {
```
Comment (MKiperszmid) [inline app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/DetailScreen.kt:224]: I personally prefer having 3 screens, one for each event. Else, you'll have to add a ton of if's to add certain parts depending on the AgendaItem

It's a bit of code duplication, but in the long run it's way more scalable

Imagine if you add extra stuff to Reminder, you would need to edit this whole screen, and would need to verify it won't break the detail screen on Task and Event also

File: core/src/main/java/com/artemissoftware/core/util/SafeLet.kt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package com.artemissoftware.core.util
+
+inline fun <T1: Any, T2: Any, R: Any> safeLet(p1: T1?, p2: T2?, block: (T1, T2)->R?): R? {
+    return if (p1 != null && p2 != null) block(p1, p2) else null
+}
```
Comment (MKiperszmid) [inline core/src/main/java/com/artemissoftware/core/util/SafeLet.kt:5]: What's the use of this?

**Review by MKiperszmid — APPROVED** [review]

File: app/src/main/java/com/artemissoftware/tasky/agenda/composables/assignment/AssignmentNotification.kt [hunk 5]
```diff
@@ -13,71 +13,78 @@ import com.artemissoftware.core.presentation.composables.icon.TaskyIcon
...
 import com.artemissoftware.tasky.R
+import com.artemissoftware.tasky.agenda.domain.models.Notification
 import com.artemissoftware.tasky.util.VisibilityTransitions
 
 @Composable
 fun AssignmentNotification(
     description: String,
-    isEditing: Boolean = false,
+    onNotificationSelected: (Notification) -> Unit,
+    notificationOptions: List<Notification>,
```
[the remark this answers is not in the document]
Comment (ArtemisSoftware) [reply app/src/main/java/com/artemissoftware/tasky/agenda/composables/assignment/AssignmentNotification.kt:24]: This contains the notification id and description

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/DetailScreen.kt [hunk 6]
```diff
@@ -0,0 +1,405 @@
...
+@Composable
+fun DetailScreen(/* TODO :Add view model here*/) {
+
+    /* TODO :Add content here*/
+}
+
+@Composable
+private fun DetailScreenContent(
+    state: DetailState,
+    events: (DetailEvents) -> Unit
```
[the remark this answers is not in the document]
Comment (ArtemisSoftware) [reply app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/DetailScreen.kt:44]: Like I have been doing on every screen file, the content composable does not have modifiers (check LoginScreen)

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/DetailState.kt (hunk 1 above)
Comment (ArtemisSoftware) [reply app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/DetailState.kt:11]: This one contains information about colors specific to the item that does not come from the AgendaItem

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/DetailScreen.kt (hunk 3 above)
Comment (ArtemisSoftware) [reply app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/DetailScreen.kt:224]: Will refactor on the next PR

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/DetailState.kt (hunk 2 above)
Comment (ArtemisSoftware) [reply app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/DetailState.kt:16]: refactored

File: core/src/main/java/com/artemissoftware/core/util/SafeLet.kt (hunk 4 above)
Comment (ArtemisSoftware) [reply core/src/main/java/com/artemissoftware/core/util/SafeLet.kt:5]: I only want to load data when I have the AgendaItem and AgendaItemType

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/DetailScreen.kt [hunk 7]
```diff
@@ -0,0 +1,405 @@
...
+) {
+
+    val context = LocalContext.current
+    val resources = context.resources
+
+
+    val title = remember(state.isEditing) {
+
+        val screenTitle = state.agendaItemType?.let{
```
[the remark this answers is not in the document]
Comment (ArtemisSoftware) [reply app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/DetailScreen.kt:53]: Noted

**Pull request merged**